src/main/java/com/spotify/docker/client/messages/swarm/Service.java

```diff
@@ -71,10 +71,37 @@
     @Nullable
     public abstract String serviceName();
 
+    /**
+     * Filter by label.
+     */
+    @Nullable
+    public abstract Label label();
+    
     public static Builder builder() {
       return new AutoValue_Service_Criteria.Builder();
     }
 
+    @AutoValue
+    public abstract static class Label {
+      
+      public abstract String name();
+      
+      public abstract String value();
+      
+      public static Builder builder() {
+        return new AutoValue_Service_Criteria_Label.Builder();
+      }
+      
+      @AutoValue.Builder
+      public abstract static class Builder {
```

Comment:
from a first glance it doesn't seem so trivial to address it (ImmutableMap doesn't take null values)